src/command_modules/azure-cli-cognitiveservices/azure/cli/command_modules/cognitiveservices/custom.py

```diff
@@ -18,6 +18,23 @@ def list_resources(client, resource_group_name=None):
     return client.list()
 
 
+def list_usages(client, resource_group_name, account_name):
+    return client.get_usages(resource_group_name, account_name).value
+
+
+def list_kinds(client):
+    """
+    List all valid kinds for Azure Cognitive Services account.
+
+    :param client: the ResourceSkusOperations
+    :return: a list
+    """
+    # The client should be ResourceSkusOperations, and list() should return a list of SKUs for all regions.
+    # The sku will have "kind" and we use that to extract full list of kinds.
+    kinds = set([x.kind for x in client.list()])
+    return sorted(list(kinds))
```

Comment:
Why is this not just an enum in the Swagger?

Reply:
1. The list will be different for different subscriptions.
2. Just like the list of regions/skus, we will update kinds list from time to time.
3. In future, we might support list kinds with filters (something like `list-kinds --location westus`).